### PR TITLE
fix(medusa): CartService lost shipping address when using the id

### PR DESCRIPTION
**Pull request by adrien2p**
**Fixes**
Proper fix to either  use the shipping_address_id or the shipping_address or in last resort the country from the region if there is only one of it

### CODE REVIEW

**Comment by adrien2p** [comment]
@dwene found the issue while looking at previous version. I will do a better fix this day

File: packages/medusa/src/services/__tests__/cart.js [hunk 1]
```diff
@@ -1,9 +1,11 @@
 import _ from "lodash"
+import { getManager } from "typeorm"
```
Comment (srindom) [inline packages/medusa/src/services/__tests__/cart.js:2]: todo: delete

File: packages/medusa/src/services/__tests__/cart.js [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
 import { MedusaError } from "medusa-core-utils"
 import { IdMap, MockManager, MockRepository } from "medusa-test-utils"
 import CartService from "../cart"
 import { InventoryServiceMock } from "../__mocks__/inventory"
 import { LineItemAdjustmentServiceMock } from "../__mocks__/line-item-adjustment"
+import { CartRepository } from "../../repositories/cart"
```
Comment (srindom) [inline packages/medusa/src/services/__tests__/cart.js:8]: todo: delete

**Review by srindom — CHANGES_REQUESTED** [review]
Small fixes otherwise looks good :)

File: packages/medusa/src/services/__tests__/cart.js [hunk 3]
```diff
@@ -262,7 +276,7 @@ describe("CartService", () => {
       expect(cartRepository.save).toHaveBeenCalledTimes(1)
     })
 
-    it("creates a cart with a prefilled shipping address", async () => {
+    it("creates a cart with a prefilled shipping address but a country not part of the region", async () => {
```
Comment (olivermrbl) [inline packages/medusa/src/services/__tests__/cart.js:279]: question: We should not be able to create a cart with a country not part of region. Is this incorrect naming of the test?

Comment (olivermrbl) [reply packages/medusa/src/services/__tests__/cart.js:279]: Should throw `Shipping country not in region`

Comment (adrien2p) [reply packages/medusa/src/services/__tests__/cart.js:279]: yes the name should include the trow well spotted :)

**Review by srindom — APPROVED** [review]
LGTM!

**Pull request merged**